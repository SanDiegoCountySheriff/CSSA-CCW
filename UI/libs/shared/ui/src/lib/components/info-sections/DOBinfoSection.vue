<template>
  <v-container class="DOB-info-container rounded">
    <v-banner class="font-weight-bold text-xl text-left mb-5">
      {{ $t('Birth Information: ') }}
    </v-banner>
    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Birth Date') }}
          </v-chip>
          <div class="info-text">
            {{ props.DOBInfo.birthDate }}
          </div>
        </div>
      </v-col>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Birth City') }}
          </v-chip>
          <div class="info-text">
            {{ props.DOBInfo.birthCity }}
          </div>
        </div>
      </v-col>
    </v-row>
    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Birth State') }}
          </v-chip>
          <div class="info-text">
            {{ props.DOBInfo.birthState }}
          </div>
        </div>
      </v-col>
      <v-col
        cols="12"
        lg="6"
      >
        <div class="info-row">
          <v-chip
            class="font-weight-bold"
            label
            :color="props.color"
          >
            {{ $t('Birth Country') }}
          </v-chip>
          <div class="info-text">
            {{ props.DOBInfo.birthCountry }}
          </div>
        </div>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import { DOBType } from '@shared-utils/types/defaultTypes';

interface IDOBInfoSectionProps {
  // eslint-disable-next-line vue/prop-name-casing
  DOBInfo: DOBType;
  color: string;
}

const props = defineProps<IDOBInfoSectionProps>();
</script>

<style lang="scss" scoped>
.DOB-info-container {
  width: 80%;
  height: 100%;
  margin: 0;
  padding: 0;
}

.info-row {
  display: flex;
  flex-direction: row;
  max-height: 2vh;
  min-height: 1vh;
}

.info-text {
  margin-left: 0.5rem;
  text-align: start;
  height: 6.5em;
  width: 50%;
  margin-bottom: 0.5rem;
  padding-left: 0.5rem;
  padding-top: 0.5rem;
  background-color: rgba(211, 241, 241, 0.3);
  border-bottom: 1px solid #666;
  border-radius: 5px;
  font-size: 1.2em;
  font-weight: bold;
}
</style>
