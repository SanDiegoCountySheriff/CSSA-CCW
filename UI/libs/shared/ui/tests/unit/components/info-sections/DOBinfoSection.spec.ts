/* eslint-disable @typescript-eslint/ban-ts-comment */
import DOBinfoSection from '@shared-ui/components/info-sections/DOBinfoSection.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('DOBinfoSection', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(DOBinfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        DOBInfo: {
          birthDate: '01/01/01',
          birthCity: 'city',
          birthState: 'state',
          birthCountry: 'country',
          currentAge: 'age',
        },
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
