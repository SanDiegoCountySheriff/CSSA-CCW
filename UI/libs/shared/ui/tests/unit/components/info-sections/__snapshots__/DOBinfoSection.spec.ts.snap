// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOBinfoSection Should match the snapshot 1`] = `
<div class="container DOB-info-container rounded">
  <transition-stub name="expand-transition" class="font-weight-bold text-xl text-left mb-5">
    <div class="v-banner v-sheet theme--light v-banner--is-mobile">
      <div class="v-banner__wrapper">
        <div class="v-banner__content">
          <div class="v-banner__text">
            Birth Information:
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Birth Date
        </span></span>
        <div class="info-text">
          01/01/01
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Birth City
        </span></span>
        <div class="info-text">
          city
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Birth State
        </span></span>
        <div class="info-text">
          state
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Birth Country
        </span></span>
        <div class="info-text">
          country
        </div>
      </div>
    </div>
  </div>
</div>
`;
